pkg/ai/iai.go

```diff
@@ -52,8 +55,9 @@ type AIConfiguration struct {
 type AIProvider struct {
 	Name     string `mapstructure:"name"`
 	Model    string `mapstructure:"model"`
-	Password string `mapstructure:"password"`
-	BaseURL  string `mapstructure:"baseurl"`
+	Password string `mapstructure:"password" yaml:"password,omitempty"`
+	BaseURL  string `mapstructure:"baseurl" yaml:"baseurl,omitempty"`
+	Engine   string `mapstructure:"engine" yaml:"engine,omitempty"`
```

Comment:
What is engine?

Reply:
Can't we just map `Model` to this?

Reply:
I believe not, model is still needed in chat completions' request.

Reply:
Perhaps it would make sense in this very custom case to have a unique struct that uses the IAI interface?

Reply:
I get your point and tbf I thought it was a small addition/change in order to need a new struct whilst working with meta tags there's no trace of engine in user's config file unless it's set by them.
Said that, if it's not readable I can follow up with another implementation of the interface.